src/test_l1_dsHdmiIn.c

```diff
@@ -2564,6 +2565,113 @@ void test_l1_dsHdmiIn_negative_dsGetEdid2AllmSupport_sink(void) {
+            UT_ASSERT_KVP_EQUAL_PROFILE_UINT32(version, keyString);
+        }
+    } else if (gSourceType == 1) {
+       // Step 4: Call dsGetHdmiVersion() with valid ports
+       UT_ASSERT_EQUAL(dsGetHdmiVersion(dsHDMI_IN_PORT_0, &version), dsERR_OPERATION_NOT_SUPPORTED);
+    }
+
+    // Step 5: Call dsHdmiInTerm() to ensure deinitialization
+    UT_ASSERT_EQUAL_FATAL(dsHdmiInTerm(), dsERR_NONE);
+
+    UT_LOG("\n Out %s\n", __FUNCTION__);
+}
+
+/**
+ * @brief Ensure dsGetHdmiVersion() returns correct error codes during negative scenarios.
+ *
+ * **Test Group ID:** Basic: 01@n
+ * **Test Case ID:** 052@n
+ *
+ * **Dependencies:** None@n
+ * **User Interaction:** None
+ *
+ * **Test Procedure:**@n
+ * |Variation / Step|Description|Test Data|Expected Result|Notes|
+ * |:--:|-----------|----------|--------------|-----|
+ * |01|Call dsGetHdmiVersion() without initializing the HDMI input sub-system |dsHDMI_IN_PORT_0, dsHdmiMaxCapabilityVersion_t*| dsERR_NOT_INITIALIZED | Should Pass |
+ * |02|Initialize the HDMI input sub-system using dsHdmiInInit() | | dsERR_NONE | Should Pass |
+ * |03|Call dsGetHdmiVersion() with invalid inputs |dsHDMI_IN_PORT_MAX, dsHdmiMaxCapabilityVersion_t*| dsERR_INVALID_PARAM | Should Pass |
+ * |04|Call dsGetHdmiVersion() with invalid inputs |dsHDMI_IN_PORT_0, NULL| dsERR_INVALID_PARAM | Should Pass |
+ * |05|Call dsHdmiInTerm() to ensure deinitialization | | dsERR_NONE | Clean up after test |
+ * |06|Call dsGetHdmiVersion() without initializing the HDMI input sub-system |dsHDMI_IN_PORT_0, dsHdmiMaxCapabilityVersion_t*| dsERR_NOT_INITIALIZED | Should Pass |
+ *
+ * @note Testing for the `dsERR_OPERATION_NOT_SUPPORTED` and `dsERR_OPERATION_FAILED` might be challenging since it requires a specific scenario where the attempted operation is not supported.
+ *
+ */
+void test_l1_dsHdmiIn_negative_dsGetHdmiVersion(void) {
+
+    gTestID = 52;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+    dsHdmiMaxCapabilityVersion_t version = HDMI_COMPATIBILITY_VERSION_MAX;
+
+    // Step 1: Call dsGetHdmiVersion() without initializing the HDMI input sub-system
+    dsError_t result = dsGetHdmiVersion(dsHDMI_IN_PORT_0, &version);
+    CHECK_FOR_EXTENDED_ERROR_CODE(result, dsERR_NOT_INITIALIZED, dsERR_NONE);
+
+    // Step 2: Initialize the HDMI input sub-system using dsHdmiInInit()
+    UT_ASSERT_EQUAL_FATAL(dsHdmiInInit(), dsERR_NONE);
+
+    if (gSourceType == 0) {
+        // Step 3: Call dsGetHdmiVersion() with invalid inputs (dsHDMI_IN_PORT_MAX)
+        UT_ASSERT_EQUAL(dsGetHdmiVersion(dsHDMI_IN_PORT_MAX, &version), dsERR_INVALID_PARAM);
+
+        // Step 4: Call dsGetHdmiVersion() with invalid inputs (NULL)
+        UT_ASSERT_EQUAL(dsGetHdmiVersion(dsHDMI_IN_PORT_0, NULL), dsERR_INVALID_PARAM);
+    } else if (gSourceType == 1) {
+        UT_ASSERT_EQUAL(dsGetHdmiVersion(dsHDMI_IN_PORT_MAX, &version), dsERR_OPERATION_NOT_SUPPORTED);
```

Comment:
let's check for NULL as well for source devices.

Reply:
Added NULL check for source